### ISSUE TITLE
Build failure: python311Packages.tables

### ISSUE THREAD

**Issue by karb94** [NixOS/nixpkgs#284146]
### Steps To Reproduce

```
{
  inputs = { nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable"; };
  outputs = { self, nixpkgs, ... }:
    let
      system = "x86_64-linux";
      pkgs = nixpkgs.legacyPackages.${system};
      pythonPkgs = p: [ p.tables ];
      myPython = pkgs.python3.withPackages pythonPkgs;
    in {
      devShells.${system}.default = pkgs.mkShell { buildInputs = [ myPython ]; };
    };
}
```

### Build log

```
error: builder for '/nix/store/4mgnlbpgs56x26ss25620i9yqmpr0fza-python3.11-tables-3.9.2.drv' failed with exit code 1;
       last 10 log lines:
       > patching file tables/utilsextension.pyx
       > Reversed (or previously applied) patch detected!  Assume -R? [n]
       > Apply anyway? [n]
       > Skipping patch.
       > 5 out of 5 hunks ignored -- saving rejects to file tables/utilsextension.pyx.rej
       > patching file tables/vlarray.py
       > Reversed (or previously applied) patch detected!  Assume -R? [n]
       > Apply anyway? [n]
       > Skipping patch.
       > 2 out of 2 hunks ignored -- saving rejects to file tables/vlarray.py.rej
       For full logs, run 'nix log /nix/store/4mgnlbpgs56x26ss25620i9yqmpr0fza-python3.11-tables-3.9.2.drv'.
error: 1 dependencies of derivation '/nix/store/f94nw2iabr0wb76bcnf3dri9i33pi4va-python3-3.11.7-env.drv' failed to build
error: 1 dependencies of derivation '/nix/store/v5b7v86xlj5aiqz613wgcr2i9kzm8ay0-nix-shell-env.drv' failed to build
```

### Additional context
I was trying to build `python311Packages.pandas-stubs` which depends on this package.

### Notify maintainers
@drewrisinger

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.0, NixOS, 24.05 (Uakari), 24.05.20240121.612f972`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by dotlambda** [comment]
This is how far I've gotten: https://github.com/NixOS/nixpkgs/compare/master...dotlambda:nixpkgs:tables-fix
The build now fails with
```
  * Copying blosc2 runtime library to 'tables' dir because it was not found in standard locations
Traceback (most recent call last):
  File "/nix/store/naw86kmy92335k30fv11cfyydxbicva8-python3.11-pyproject-hooks-1.0.0/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
    main()
  File "/nix/store/naw86kmy92335k30fv11cfyydxbicva8-python3.11-pyproject-hooks-1.0.0/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 335, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/naw86kmy92335k30fv11cfyydxbicva8-python3.11-pyproject-hooks-1.0.0/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
    return _build_backend().build_wheel(wheel_directory, config_settings,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/dyl7f7v3cfi7j9d4z42pn2ss0bgw0r33-python3.11-setuptools-69.0.2/lib/python3.11/site-packages/setuptools/build_meta.py", line 404, in build_wheel
    return self._build_with_temp_dir(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/dyl7f7v3cfi7j9d4z42pn2ss0bgw0r33-python3.11-setuptools-69.0.2/lib/python3.11/site-packages/setuptools/build_meta.py", line 389, in _build_with_temp_dir
    self.run_setup()
  File "/nix/store/dyl7f7v3cfi7j9d4z42pn2ss0bgw0r33-python3.11-setuptools-69.0.2/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
    exec(code, locals())
  File "<string>", line 919, in <module>
  File "/nix/store/w4fvvhkzb0ssv0fw5j34pw09f0qw84w8-python3-3.11.7/lib/python3.11/shutil.py", line 419, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/nix/store/w4fvvhkzb0ssv0fw5j34pw09f0qw84w8-python3-3.11.7/lib/python3.11/shutil.py", line 258, in copyfile
    with open(dst, 'wb') as fdst:
         ^^^^^^^^^^^^^^^
PermissionError: [Errno 13] Permission denied: '/build/tables-3.9.2/tables/libblosc2.so'
```
cc @tjni @natsukium @benxiao @mweinelt @rmcgibbo

**Comment by karb94** [comment]
The package builds without issues now. Possibly fixed in #293429.
